File: cham/progate/Ruby_5/food.rb
require "./menu.rb"

class Food < Menu
  # Menuのインスタンスに加えてcalorieを追加
  attr_accessor :calorie

  def initialize(name:, price:, calorie:)
    super(name: name, price: price)
    self.calorie = calorie
  end

  def info
    "#{self.name} #{self.price}円 (#{self.calorie}kcal)"
  end
  
  def calorie_info
    "#{self.name}は#{self.calorie}kcalです"
  end
end

File: cham/progate/Ruby_5/food.js
import { Menu } from "./menu";

class Food extends Menu {
  constructor(name, price, calorie) {
    super(name, price);
    this.calorie = calorie;
  }

  info() {
    return `${this.name} ${this.price}円 (${this.calorie}kcal)`;
  }

  calorie_info() {
    return `${this.name}は(${this.calorie}kcal)`;
  }
}
